benchmarks/infill_density/infill_ascii_data.cc

```diff
@@ -0,0 +1,262 @@
+    }
+
+
+    template <int dim>
+    Tensor<1,dim>
+    InfillAsciiData<dim>::
+    boundary_traction (const types::boundary_id surface_boundary_id,
+                       const Point<dim> &position,
+                       const Tensor<1,dim> &normal_vector) const
+    {
+      const double load = Utilities::AsciiDataBoundary<dim>::get_data_component(surface_boundary_id,
+                                                                                position,
+                                                                                0);
+      Tensor<1, dim> traction = normal_vector;
+      const double gravity_norm = this->get_gravity_model().gravity_vector(position).norm();
+      const double elevation = this->get_geometry_model().height_above_reference_surface(position);
+
+      // Check if the load at the current point has a height which exceeds the
+      // 'rock_infill height'. The assumption is that the values taken from the
+      // ASCII file are specifying the part of the load that is above the undeformed
+      // reference surface. If we consider a seamount as the load, the ASCII
+      // file takes the bathymetry of the seamount and provides it as a traction.
+      // However, the volcanic rock comprising the seamount also fills in the
+      // flexural moat, providing an additional traction. Since it is much harder to
+      // specify this traction, 'rock_infill_height' determines where the rock_density
+      // that the seamount is made of will infill the flexural moat, and where
+      // sediment_density will infill the flexural moat. For this test, this
+      // variable is not that important given the idealized load, but this is useful
+      // when using more complicated bathymetry maps where small scale seafloor
+      // features makes it unrealistic to set rock_infill_height=0.
+      if (load >= rock_infill_height*gravity_norm*rock_density)
+        {
+          traction = (-load + elevation*gravity_norm*rock_density) * normal_vector;
+        }
+      else
+        {
+          traction = (-load + elevation*gravity_norm*sediment_density) * normal_vector;
+        }
+      return traction;
+    }
+
+
+    template <int dim>
+    void
+    InfillAsciiData<dim>::update()
+    {
+      Interface<dim>::update ();
+      Utilities::AsciiDataBoundary<dim>::update();
+    }
+
+
+    template <int dim>
+    void
+    InfillAsciiData<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Boundary traction model");
+      {
+        Utilities::AsciiDataBoundary<dim>::declare_parameters(prm,
+                                                              "$ASPECT_SOURCE_DIR/tests/infill_density/",
+                                                              "triangle_load.txt");
+        prm.enter_subsection("Infill ascii");
+        {
+          prm.declare_entry ("Rock density", "2800",
+                             Patterns::List(Patterns::Double(0.)),
```

Comment:
Could this and the patterns below just be `Patterns::Double (0.)`? In other words, the plugin is currently designed to only use one value for each and they are declared as doubles (rather than vectors of doubles).

Reply:
Ah yes good catch!